src/Lunr/Core/Configuration.php

```diff
@@ -158,6 +162,42 @@ public function load_file(string $identifier): void
 
         $this->config      = $config;
         $this->sizeInvalid = TRUE;
+
+        $this->loaded[] = $identifier;
+    }
+
+    /**
+     * Attempt to autoload a config file.
+     *
+     * @param mixed $identifier Identifier string for the config file to load.
+     *                          e.g.: For conf.lunr.inc.php the identifier would be 'lunr'
+     *
+     * @return void
+     */
+    private function autoloadFile(mixed $identifier): void
+    {
+        if (!$this->isRootConfig)
+        {
+            return;
+        }
+
+        if (!is_string($identifier))
+        {
+            return;
+        }
```

Comment:
Why would we allow this?

Reply:
integers are valid array keys, and we can't just not allow them because they would still be valid as (sub) config keys.

It will hardly ever trigger in actual situations, but phpstan needs it to ascertain type safety for the later code.